tree/bst/bst.py

```diff
@@ -0,0 +1,119 @@
+"""
+bst
+"""
+
+import unittest
+
+class Node(object):
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+    """
+        Insert data in bst
+        Using recursion. Complexity O(logN)
+    """
+    def insert(self, data):
+        if self.data == data:      # The data is already there
+            return False
+        elif data < self.data:     # Go to left root
+            if self.left:          # If left root is a node
+                return self.left.insert(data)
+            else:                  # left root is a None
+                self.left = Node(data)
+                return True
+        else:                      # Go to right root
+            if self.right:         # If right root is a node
+                return self.right.insert(data)
+            else:
+                self.right = Node(data)
+                return True
+    """
+        Search data in bst
+        Using recursion. Complexity O(logN)
+    """
+    def search(self, data):
+        if self.data == data:
+            return True
+        elif self.left is None and self.right is None:
+            return False
+        elif data > self.data:     # Go to right root
+            return self.right.search(data)
+        else:                      # Go to left root
+            return self.left.search(data)
+    """
+        Get the number of elements
+        Using recursion. Complexity O(logN)
+    """
+    def size(self):
```

Comment:
these methods must be defined inside `BST` class, not inside `Node`.

Reply:
What do you mean?